scripts/release-branch.js

```diff
@@ -0,0 +1,121 @@
+// Added shelljs to use maybe in the script "npm install shelljs"
+// https://github.com/shelljs/shelljs
+const shell = require('shelljs');
+
+// Added semver to use for increment the version "npm install semver"
+// https://github.com/npm/node-semver
+const semver = require('semver');
+
+// Used for the file system for changelog.
+const path = require('path');
+const fs = require('fs');
+const yaml = require('js-yaml');
+
+// Find out the latest release tag and display it in the command line
+const latest = shell.exec('git describe --abbrev=0 --tags');
+
+// Display the latest tag.
+shell.echo('Display the current tag:', latest);
+
+// Increment the release branch.
+const minor = semver.inc(latest.toString(), 'minor');
+
+// GitHub API
+const https = require('https')
+
+// Print out the new minor version
+shell.echo(minor);
+
+// Update the changelog.md file
+const today = new Date();
+const day = today.getDay();
+const month = today.getMonth();
+const year = today.getFullYear();
+
+// Changelog.md title for each release
+const title = `## ${minor} (${month}/${day}/${year})`;
+// Look at the changelog files
+
+const directoryPath = path.resolve(__dirname, '../changelogs');
+const changelogPath = `${path.resolve(__dirname, '../')}/CHANGELOG.md`;
+
+let newLogs = [];
+// Read directory path and exclude the template.yml file.
+const changelogs = fs.readdirSync(directoryPath).filter(function(template) {
+  return template !== 'template.yml';
+});
+changelogs.forEach((fileName) => {
+  const content = yaml.safeLoad(fs.readFileSync(`${directoryPath}/${fileName}`, 'utf8'));
+  Object.keys(content).forEach((changeType, i) => {
+  	newLogs.push(`\n### ${changeType} \n`)
+  	content[changeType].forEach((change) => {
+  		const newChange = `- (${change.project}) [${change.component}] ${change.issue}: ${change.description}\n`
+  		newLogs.push(newChange)
+  	})
+  });
+});
+
+const fd = fs.readFileSync(changelogPath).toString().split("\n");
+fd.splice(0, 0, title, newLogs.join(''));
+var allLogs = fd.join('\n');
+
+// Remove the changelog files
+fs.readdir(directoryPath, function(err, items) {
+  for (var i=0; i<items.length; i++) {
+    // console.log(items[i]);
+    if (items[i] != "template.yml") {
+      fs.unlink(items, (err) => {
```

Comment:
You're inside a loop - you should just use `i` here to specify the item you want:
```javascript
for (var i=0; i<items.length; i++) {
    if (items[i] != "template.yml") {
      fs.unlink(item[1], (err) => {
        if(err) throw err;
      });
    }
}
```
Also, watch out here, cause `fs.readdir` can throw an error as well. So before you start looping over items, check that `err` and throw it if there is one. Also, one day soon, let's talk about promises. They make this kind of thing a lot easier.

Reply:
you can also use `fs.readdirSync` it will run this synchronous